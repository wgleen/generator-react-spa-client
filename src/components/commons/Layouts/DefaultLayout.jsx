import React from 'react'
import PropTypes from 'prop-types'

const DefaultLayout = (props) => {
  const { children } = props

  return (
    <div>
      <section>{children}</section>
    </div>
  )
}

DefaultLayout.defaultProps = {
  children: null
}

DefaultLayout.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.element)
  ])
}

export default DefaultLayout
