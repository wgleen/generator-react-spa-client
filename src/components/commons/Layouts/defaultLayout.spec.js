import React from 'react'
import { mount } from 'enzyme'
import DefaultLayout from './DefaultLayout'

describe('<DefaultLayout />', () => {
  it('should render correctly', () => {
    const component = mount(<DefaultLayout />)

    expect(component).toMatchSnapshot()

    component.unmount()
  })

  it('should render the children prop given', () => {
    const children = 'Text children'

    const component = mount(<DefaultLayout>{children}</DefaultLayout>)

    expect(component.text()).toEqual(children)

    component.unmount()
  })
})
