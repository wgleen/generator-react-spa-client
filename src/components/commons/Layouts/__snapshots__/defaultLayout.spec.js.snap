// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DefaultLayout /> should render correctly 1`] = `
<DefaultLayout>
  <div>
    <section />
  </div>
</DefaultLayout>
`;
